File: PasswordValidation/PasswordValidation/CompositePasswordValidator.cs
namespace PasswordValidation;

public class CompositePasswordValidator : IPasswordValidator
{
    private readonly IEnumerable<IPasswordValidator> _validators;

    public CompositePasswordValidator(IEnumerable<IPasswordValidator> validators)
    {
        _validators = validators;
    }

    public bool Validate(string password)
    {
        foreach (var validator in _validators)
        {
            if (!validator.Validate(password))
            {
                return false;
            }
        }

        return true;
    }
}

File: PasswordValidation/py/password_validation/composite_password_validator.py
from password_validation.flexible_password_validator import PasswordValidator


class CompositePasswordValidator(PasswordValidator):
    def __init__(self, validators: list):
        self.__validators = validators

    def validate(self, password: str) -> bool:
        for validator in self.__validators:
            if not validator.validate(password):
                return False
        return True
